### PR TITLE
Add tests for sycl::atomic_ref static members

### PR DESCRIPTION
**Pull request by vladimirkhashev**
Create tests according to 3.1.1. Members section of test plan https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/atomic_ref.asciidoc

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

File: tests/atomic_ref/atomic_ref_common.h [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                                                     sycl::memory_order order,
+                                                     sycl::memory_scope scope) {
+  return !memory_order_and_scope_are_supported(q, order, scope);
+}
+
+/**
+ * @brief Function to compare two floatinig point values
+ */
+template <typename T>
+bool compare_act_and_expd_with_epsilon(T actual, T expected, T eps) {
```
[the remark this answers is not in the document]
Comment (vladimirkhashev) [reply tests/atomic_ref/atomic_ref_common.h]: Changed.

File: tests/atomic_ref/atomic_ref_common.h [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+inline bool memory_order_and_scope_are_not_supported(sycl::queue& q,
+                                                     sycl::memory_order order,
+                                                     sycl::memory_scope scope) {
+  return !memory_order_and_scope_are_supported(q, order, scope);
+}
+
+/**
+ * @brief Function to compare two floatinig point values
```
[the remark this answers is not in the document]
Comment (vladimirkhashev) [reply tests/atomic_ref/atomic_ref_common.h]: Fixed.

File: tests/common/type_coverage.h [hunk 3]
```diff
@@ -595,4 +609,5 @@ void for_all_types_and_marrays(const named_type_pack<types...> &typeList,
     ++typeNameIndex),
    ...);
 }
-#endif  // __SYCLCTS_TESTS_COMMON_TYPE_COVERAGE_H
+#endif  // !SYCL_CTS_COMPILING_WITH_HIPSYCL
+#endif  // __SYCLCTS_TESTS_COMMON_TYPE_COVERAGE_H
```
Comment (bader) [inline tests/common/type_coverage.h:613]:
```suggestion
#endif  // __SYCLCTS_TESTS_COMMON_TYPE_COVERAGE_H

```

File: tests/atomic_ref/atomic_ref_static_members_core.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+// FIXME: re-enable when sycl::access::address_space::generic_space is
+// implemented in computecpp
+DISABLED_FOR_TEST_CASE(ComputeCpp)
+("sycl::atomic_ref static members. core types", "[atomic_ref]")({
+  const auto types = atomic_ref::tests::common::get_conformance_type_pack();
+  for_all_types<atomic_ref::static_members::run_test>(types);
+});
+
+}  // namespace atomic_ref::static_members::core
```
Comment (bader) [inline tests/atomic_ref/atomic_ref_static_members_core.cpp:41]:
```suggestion
}  // namespace atomic_ref::static_members::core

```

File: tests/atomic_ref/atomic_ref_static_members_core_atomic64.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace atomic_ref::static_members::core::atomic64 {
+
+// FIXME: re-enable when sycl::access::address_space::generic_space is
+// implemented in computecpp
+DISABLED_FOR_TEST_CASE(ComputeCpp)
+("sycl::atomic_ref static members. atomic64 types", "[atomic_ref]")({
+  auto queue = sycl_cts::util::get_cts_object::queue();
+  if (!queue.get_device().has(sycl::aspect::atomic64)) {
+    WARN(
```
Comment (bader) [inline tests/atomic_ref/atomic_ref_static_members_core_atomic64.cpp:39]: WARN -> SKIP

**Review by bader — APPROVED** [review]
LGTM, a couple of minor suggestions.

File: tests/atomic_ref/atomic_ref_static_members_core.cpp (hunk 4 above)
Comment (vladimirkhashev) [reply tests/atomic_ref/atomic_ref_static_members_core.cpp]: Done

File: tests/common/type_coverage.h (hunk 3 above)
Comment (vladimirkhashev) [reply tests/common/type_coverage.h]: Done

File: tests/atomic_ref/atomic_ref_static_members_core_atomic64.cpp (hunk 5 above)
Comment (vladimirkhashev) [reply tests/atomic_ref/atomic_ref_static_members_core_atomic64.cpp]: Done

**Review by keryell — APPROVED** [review]

**Pull request merged**